src/expander.wisp

```diff
@@ -335,3 +345,35 @@
   [& body]
   `(.call lazy-seq nil false (fn [] ~@body)))
 (install-macro :lazy-seq expand-lazy-seq)
+
+
+(defn expand-when
+  "Evaluates test. If logical true, evaluates body in an implicit do."
+  [test & body]
+  `(if ~test (do ~@body)))
+(install-macro :when expand-when)
+
+(defn expand-when-not
+  "Evaluates test. If logical false, evaluates body in an implicit do."
+  [test & body]
+  `(when (not ~test) ~@body))
+(install-macro :when-not expand-when-not)
+
+
+(defn expand-if-let
+  "bindings => binding-form test
+  body => [then else]
+  If test is true, evaluates then with binding-form bound to the value of
+  test, if not, yields else."
+  [bindings then else*]
+  (let [name (first bindings), test (second bindings), sym (gensym)]
```

Comment:
for slightly better debugging, maybe pass `(first bindings)` to `gensym` so the name is better?

Reply:
I considered that, but didn't think it mattered much. Still, I suppose it would be indeed an improvement…